code/modules/surgery/organs/blood.dm

```diff
@@ -117,264 +73,218 @@
-			if(donor_antigen != "A" && donor_antigen != "O") return 1
-		if("B")
-			if(donor_antigen != "B" && donor_antigen != "O") return 1
-		if("O")
-			if(donor_antigen != "O") return 1
-		//AB is a universal receiver.
-	return 0
-
-/*
-Target: Thing/tile to get bloody
-Source: Human or blood reagent
-Large: Whether the splat should be big or not
-*/
-/proc/blood_splatter(var/target,var/source,var/large = 0)
-
-	var/obj/effect/decal/cleanable/blood/B
-	var/decal_type = /obj/effect/decal/cleanable/blood/splatter
-	var/turf/T = get_turf(target)
-	var/datum/reagent/blood/bld
-	if(istype(source,/mob/living/carbon/human))
-		var/mob/living/carbon/human/H = source
-		bld = H.get_blood(H.vessel)
-		if(H.species.exotic_blood)
-			H.vessel.reaction(T, TOUCH)
-			return
-		else if(H.species.flags & NO_BLOOD)
+// This is has more potential uses, and is probably faster than the old proc.
+/proc/get_safe_blood(bloodtype)
+	. = list()
+	if(!bloodtype)
+		return
+	switch(bloodtype)
+		if("A-")
+			return list("A-", "O-")
+		if("A+")
+			return list("A-", "A+", "O-", "O+")
+		if("B-")
+			return list("B-", "O-")
+		if("B+")
+			return list("B-", "B+", "O-", "O+")
+		if("AB-")
+			return list("A-", "B-", "O-", "AB-")
+		if("AB+")
+			return list("A-", "A+", "B-", "B+", "O-", "O+", "AB-", "AB+")
+		if("O-")
+			return list("O-")
+		if("O+")
+			return list("O-", "O+")
+
+//to add a splatter of blood or other mob liquid.
+/mob/living/proc/add_splatter_floor(turf/T, small_drip)
+	if(get_blood_id() != "blood")//is it blood or welding fuel?
+		return
+
+	if(!T)
+		T = get_turf(src)
+
+	var/list/temp_blood_DNA
+	var/list/b_data = get_blood_data(get_blood_id())
+
+	if(small_drip)
+		// Only a certain number of drips (or one large splatter) can be on a given turf.
+		var/obj/effect/decal/cleanable/blood/drip/drop = locate() in T
+		if(drop)
+			if(drop.drips < 3)
+				drop.drips++
+				drop.overlays |= pick(drop.random_icon_states)
+				drop.transfer_mob_blood_dna(src)
+				drop.basecolor = b_data["blood_color"]
+				drop.update_icon()
+			else
+				temp_blood_DNA = list()
+				temp_blood_DNA |= drop.blood_DNA.Copy() //we transfer the dna from the drip to the splatter
+				qdel(drop)
+		else
+			drop = new(T)
+			drop.transfer_mob_blood_dna(src)
+			drop.basecolor = b_data["blood_color"]
+			drop.update_icon()
 			return
-	else if(istype(source, /datum/reagent))
-		bld = source
-		if(!istype(bld, /datum/reagent/blood))
-			var/datum/reagent/R = bld
+
+	// Find a blood decal or create a new one.
+	var/obj/effect/decal/cleanable/blood/B = locate() in T
+	if(!B)
+		B = new /obj/effect/decal/cleanable/blood/splatter(T)
+	B.transfer_mob_blood_dna(src) //give blood info to the blood decal.
+	if(temp_blood_DNA)
+		B.blood_DNA |= temp_blood_DNA
+	B.update_icon()
+
+/mob/living/carbon/human/add_splatter_floor(turf/T, small_drip)
+	if(!(species && species.flags & NO_BLOOD))
+		if(species.exotic_blood)
+			if(!T)
+				T = get_turf(src)
+			var/datum/reagent/R = get_blood_id()
```

Comment:
`var/datum/reagent/R = chemical_reagents_list[get_blood_id()]`

Otherwise you're just trying to assign a string to the variable.
This fixes slime people not bleeding. 😏